tests/providers/amazon/aws/operators/test_eks.py

```diff
@@ -0,0 +1,299 @@
+        )
+
+        self.create_cluster_operator_with_nodegroup = EKSCreateClusterOperator(
+            task_id=TASK_ID,
+            **self.create_cluster_params,
+            **self.base_nodegroup_params,
+        )
+
+    @mock.patch.object(EKSHook, "create_cluster")
+    @mock.patch.object(EKSHook, "create_nodegroup")
+    def test_execute_create_cluster(self, mock_create_nodegroup, mock_create_cluster):
+        self.create_cluster_operator.execute({})
+
+        mock_create_cluster.assert_called_once_with(**convert_keys(self.create_cluster_params))
+        mock_create_nodegroup.assert_not_called()
+
+    @mock.patch.object(EKSHook, "get_cluster_state")
+    @mock.patch.object(EKSHook, "create_cluster")
+    @mock.patch.object(EKSHook, "create_nodegroup")
+    def test_execute_when_called_with_nodegroup_creates_both(
+        self, mock_create_nodegroup, mock_create_cluster, mock_cluster_state
+    ):
+        mock_cluster_state.return_value = STATUS_VALUE
+
+        self.create_cluster_operator_with_nodegroup.execute({})
+
+        mock_create_cluster.assert_called_once_with(**convert_keys(self.create_cluster_params))
+        mock_create_nodegroup.assert_called_once_with(**convert_keys(self.create_nodegroup_params))
+
+
+class TestEKSCreateNodegroupOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
+        self.nodegroup_name: str = random_names()
+
+        self.create_nodegroup_params = dict(
+            cluster_name=self.cluster_name,
+            nodegroup_name=self.nodegroup_name,
+            nodegroup_subnets=SUBNETS_VALUE,
+            nodegroup_role_arn=NODEROLE_ARN_VALUE,
+        )
+
+        self.create_nodegroup_operator = EKSCreateNodegroupOperator(
+            task_id=TASK_ID, **self.create_nodegroup_params
+        )
+
+    @mock.patch.object(EKSHook, "create_nodegroup")
+    def test_execute_when_nodegroup_does_not_already_exist(self, mock_create_nodegroup):
+        self.create_nodegroup_operator.execute({})
+
+        mock_create_nodegroup.assert_called_once_with(**convert_keys(self.create_nodegroup_params))
+
+
+class TestEKSDeleteClusterOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
+
+        self.delete_cluster_operator = EKSDeleteClusterOperator(
+            task_id=TASK_ID, cluster_name=self.cluster_name
+        )
+
+    @mock.patch.object(EKSHook, "list_nodegroups")
+    @mock.patch.object(EKSHook, "delete_cluster")
+    def test_existing_cluster_not_in_use(self, mock_delete_cluster, mock_list_nodegroups):
+        mock_list_nodegroups.return_value = dict(nodegroups=list())
+
+        self.delete_cluster_operator.execute({})
+
+        mock_list_nodegroups.assert_called_once
+        mock_delete_cluster.assert_called_once_with(name=self.cluster_name)
+
+
+class TestEKSDeleteNodegroupOperator(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cluster_name: str = random_names()
```

Comment:
Is it needed to generate random value?  Can you use a fixed value here?

Reply:
Removed in upcoming revision.

Reply:
Corrected in https://github.com/apache/airflow/pull/16571/commits/76c7ad635948191764a323a9be03fa965fc91ff5